server/blob/blob-api/src/test/java/org/apache/james/blob/api/BlobStoreContract.java

```diff
@@ -25,138 +25,133 @@
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.UncheckedIOException;
 import java.nio.charset.StandardCharsets;
 
-import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.api.Test;
 
 import com.google.common.base.Strings;
 
 public interface BlobStoreContract {
 
+    byte[] EMPTY_BYTEARRAY = {};
+    byte[] SHORT_BYTEARRAY = "toto".getBytes(StandardCharsets.UTF_8);
+    byte[] TEN_KILOBYTES = Strings.repeat("0123456789\n", 1000).getBytes(StandardCharsets.UTF_8);
+    byte[] TWELVE_MEGABYTES = Strings.repeat("0123456789\r\n", 1024 * 1024).getBytes(StandardCharsets.UTF_8);
+
     BlobStore testee();
 
     BlobId.Factory blobIdFactory();
 
     @Test
-    default void saveShouldReturnEmptyWhenNullData() throws Exception {
+    default void saveShouldThrowWhenNullData() {
         assertThatThrownBy(() -> testee().save((byte[]) null))
             .isInstanceOf(NullPointerException.class);
     }
 
     @Test
-    default void saveShouldReturnEmptyWhenNullInputStream() throws Exception {
+    default void saveShouldThrowWhenNullInputStream() {
         assertThatThrownBy(() -> testee().save((InputStream) null))
             .isInstanceOf(NullPointerException.class);
     }
 
     @Test
-    default void saveShouldSaveEmptyData() throws Exception {
-        BlobId blobId = testee().save(new byte[]{}).join();
+    default void saveShouldSaveEmptyData() {
+        BlobId blobId = testee().save(EMPTY_BYTEARRAY).join();
 
         byte[] bytes = testee().readBytes(blobId).join();
 
         assertThat(new String(bytes, StandardCharsets.UTF_8)).isEmpty();
     }
 
     @Test
-    default void saveShouldSaveEmptyInputStream() throws Exception {
-        BlobId blobId = testee().save(new ByteArrayInputStream(new byte[]{})).join();
+    default void saveShouldSaveEmptyInputStream() {
+        BlobId blobId = testee().save(new ByteArrayInputStream(EMPTY_BYTEARRAY)).join();
```

Comment:
would it be interesting to hard code the empty blobId, so that we don't need to reach the storage for it?

Reply:
I think it doesn't happen in real world so It probably doesn't worth it

Reply:
I think it does, empty attachments are pretty common. The question is more: do we care?

Reply:
not in 

> don't rely on empty content for error handling

The test is: do you care enough to reach JIRA and open a ticket? (: